app/caps-controller-manager/controllers/metalmachine_controller.go

```diff
@@ -133,7 +132,14 @@ func (r *MetalMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	controllerutil.AddFinalizer(metalMachine, infrav1.MachineFinalizer)
 
-	// If server ref is already provided, server binding controller is going to reconcile matching server binding
+	// TODO (smira):
+	// This is really weird that .Spec.ServerRef is used both to set the manual link to the server by the user
+	// and a way to store current binding when server is picked up automatically
+	//
+	// This should be refactored to keep _current_ binding ref in the .Status, while
+	// .spec should be reserved for manually chosen server only.
+	// This opens a question what to do if the `.Status` is lost after pivoting, how to reconcile it correctly?
```

Comment:
is it possible to tell capi to migrate that over?

Reply:
that would be possible if we change the schema, for this PR I just documented the problem as it is right now

Reply:
makes me wonder if we should discuss how the whole servers and serverclasses and the binding gets populated, feels like room for improvement

Reply:
yeah, I think in Sidero 0.6 we should do more improvements as we change our schema actually